### ISSUE TITLE
Builds should fail when there are test that have failed before

### ISSUE THREAD

**Issue by doudou** [tidewise/buildbot-ci#28]
Currently, tests that are not re-run won't affect the result of the build. This means that a build can be green while there are some unit-test failing. This can be seen in the Autoproj view, but it is confusing and sometimes one does not realize it is so (and wrongly assume the build is actually green while it is not)

**Comment by g-arjones** [comment]
Yeah... This is pretty bad. Can you just make `autoproj ci build|test` return non-zero if there are failed cached builds/tests?

**Comment by doudou** [comment]
That would be the idea, yes :P
